arch/cortex-m/src/syscall.rs

```diff
@@ -47,6 +49,56 @@ pub struct CortexMStoredState {
     psp: usize,
 }
 
+const MAJOR_VER: usize = 1;
+const MINOR_VER: usize = size_of::<CortexMStoredState>();
+const TAG: [u8; 4] = [b'c', b't', b'x', b'm'];
+const METADATA_LEN: usize = 3;
+
+const USIZE_SZ: usize = size_of::<usize>();
+fn usize_byte_range(index: usize) -> Range<usize> {
+    index * USIZE_SZ..(index + 1) * USIZE_SZ
+}
+
+fn usize_from_u8_slice(slice: &[u8], index: usize) -> Result<usize, ErrorCode> {
+    let range = usize_byte_range(index);
+    if range.end < slice.len() {
+        Ok(usize::from_le_bytes(
+            slice[range].try_into().or(Err(ErrorCode::FAIL))?,
+        ))
+    } else {
+        Err(ErrorCode::SIZE)
+    }
+}
+
+fn write_usize_to_u8_slice(val: usize, slice: &mut [u8], index: usize) {
+    let range = usize_byte_range(index);
+    slice[range].copy_from_slice(&val.to_le_bytes());
+}
+
+impl core::convert::TryFrom<&[u8]> for CortexMStoredState {
+    type Error = ErrorCode;
+    fn try_from(ss: &[u8]) -> Result<CortexMStoredState, Self::Error> {
+        if ss.len() >= size_of::<CortexMStoredState>() + METADATA_LEN * USIZE_SZ
```

Comment:
One more question: why support slices larger than the size we are converting? Wouldn't passing a slice larger than the size of `CortexMStoredState` likely indicate some bug?

Reply:
Is it possible the slice would come from somewhere that doesn't know which architecture will be used? Or that it would want to keep access to a longer slice?

Reply:
It's read into a large buffer from flash and I was just passing the rest of the buffer from the start of the stored state. Updated to expect actual size.